jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -156,7 +156,7 @@ public int read(byte[] b, int off, int len) throws IOException
                 long period=System.nanoTime()-_firstByteTimeStamp;
                 if (period>=1000)
                 {
-                    double data_rate = _contentArrived / ((1.0*period)/TimeUnit.SECONDS.toNanos(1));
+                    double data_rate = _contentArrived / (((double)period)/TimeUnit.SECONDS.toNanos(1));
```

Comment:
Ah found your version and it is not good.   Will be horribly numerically sensitive for short periods much less than 1s and for periods like 1.001 vs 2.999.

let me see if I can do the maths the other way around so rather than calculate a per second number and compare, we compute a minimum data arrived and compare....